src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -282,7 +282,9 @@ export class RecommendationHandler {
             })
             this.recommendations = isCloud9() ? recommendation : this.recommendations.concat(recommendation)
             this._onDidReceiveRecommendation.fire()
-        } else {
+        }
+        // send Empty userDecision event only when all paginated request finishes
```

Comment:
The description should actually be: 
send Empty userDecision event if user receives no recommendations in this session at all.

Reply:
Updated with new description